factory/src/main/kotlin/io/qalipsis/core/factory/steps/meter/checkers/ValueCheckerConverter.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * QALIPSIS
+ * Copyright (C) 2025 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.qalipsis.core.factory.steps.meter.checkers
+
+import io.qalipsis.api.meters.Meter
+import io.qalipsis.api.meters.steps.checkSpecification.RangeValueCheckSpecification
+import io.qalipsis.api.meters.steps.checkSpecification.SingleThresholdValueCheckSpecification
+import io.qalipsis.api.meters.steps.checkSpecification.SpecificationType
+import io.qalipsis.api.meters.steps.checkSpecification.ValueCheckSpecification
+
+/**
+ * Converts a value check specification into an executable checker.
+ *
+ * @author Francisca Eze
+ */
+internal class ValueCheckerConverter {
+
+    /**
+     * @param checkSpec contains the comparison rule.
+     */
+    fun <T : Comparable<T>, M : Meter<M>> convert(
+        checkSpec: ValueCheckSpecification<T>
+    ): ValueChecker<T> {
+        return when (checkSpec.type) {
```

Comment:
Why do you need a type variable? testing the type of checkSpec would be much simpler.

return when (checkSpec) {
 is SingleThresholdValueCheckSpecification -> LessThanChecker(checkSpec.threshold)

Please remove that property from the spec interface.

Reply:
By its type, see my example.